blast/cactus_blast.py

```diff
@@ -55,232 +56,318 @@ def __init__(self, chunkSize=10000000, overlapSize=10000,
+    def __init__(self, cactusDisk, flowerName, finalResultsFile, blastOptions):
+        Job.__init__(self)
+        self.cactusDisk = cactusDisk
+        self.flowerName = flowerName
+        self.finalResultsFile = finalResultsFile
+        self.blastOptions = blastOptions
+    def run(self, fileStore):
+        resultsID = self.addChild(BlastFlower(self.cactusDisk, self.flowerName, self.blastOptions)).rv()
+        self.addFollowOn(WritePermanentFile(resultsID, self.finalResultsFile))
 
-class BlastFlower(Target):
+class BlastFlower(Job):
     """Take a reconstruction problem and generate the sequences in chunks to be blasted.
     Then setup the follow on blast targets and collation targets.
     """
-    def __init__(self, cactusDisk, flowerName, finalResultsFile, blastOptions):
-        Target.__init__(self)
+    def __init__(self, cactusDisk, flowerName, blastOptions):
+        Job.__init__(self)
         self.cactusDisk = cactusDisk
         self.flowerName = flowerName
-        self.finalResultsFile = finalResultsFile
         self.blastOptions = blastOptions
         blastOptions.roundsOfCoordinateConversion = 2
         
-    def run(self):
-        chunksDir = makeSubDir(os.path.join(self.getGlobalTempDir(), "chunks"))
+    def run(self, fileStore):
+        chunksDir = getTempDirectory(rootDir=fileStore.getLocalTempDir())
         chunks = [ chunk for chunk in popenCatch("cactus_blast_chunkFlowerSequences %s '%s' %s %i %i %i %s" % \
                                                           (getLogLevelString(), self.cactusDisk, self.flowerName, 
                                                           self.blastOptions.chunkSize, 
                                                           self.blastOptions.overlapSize,
                                                           self.blastOptions.minimumSequenceLength,
                                                           chunksDir)).split("\n") if chunk != "" ]
         logger.info("Broken up the flowers into individual 'chunk' files")
-        self.addChildTarget(MakeBlastsAllAgainstAll(self.blastOptions, chunks, self.finalResultsFile))
-        
-class BlastSequencesAllAgainstAll(Target):
+        chunkIDs = [fileStore.writeGlobalFile(chunk, cleanup=False) for chunk in chunks]
+        selfResultsID = self.addChild(MakeSelfBlasts(self.blastOptions, chunkIDs)).rv()
+        offDiagonalResultsID = self.addChild(MakeOffDiagonalBlasts(self.blastOptions, chunkIDs)).rv()
+        return self.addFollowOn(CollateBlasts([selfResultsID, offDiagonalResultsID])).rv()
+    
+class BlastSequencesAllAgainstAllWrapper(Job):
+    def __init__(self, sequenceFiles1, outputFile, blastOptions):
+        Job.__init__(self)
+        self.sequenceFiles1 = sequenceFiles1
+        self.outputFile = outputFile
+        self.blastOptions = blastOptions
+    def run(self, fileStore):
+        sequenceIDs1 = [fileStore.writeGlobalFile(seq, cleanup=False) for seq in self.sequenceFiles1]
```

Comment:
We should use the context manager and its "importFile" function rather than this wrapper, once it's ready. Otherwise it may not work correctly on "the cloud." But it's not ready yet, so this is OK.